src/go/rpk/pkg/cli/cmd/root.go

```diff
@@ -80,8 +84,37 @@ func Execute() {
 
 	addPlatformDependentCmds(fs, mgr, rootCmd)
 
-	// Plugins: we favor rpk commands first, but if we cannot find one, we
-	// try a plugin. If we exec a plugin, tryExecPlugin exits.
+	// To support autocompletion even for plugins, we list all plugins now
+	// and add tiny commands to our root command. Cobra works by creating
+	// autocompletion scripts that contain the commands discoverable from
+	// the root command, so by adding cobra.Command's for all plugins, we
+	// allow autocompletion.
+	plugins := listPlugins(fs, filepath.SplitList(os.Getenv("PATH")))
+	{
```

Comment:
There used to be more logic within this scope (temporary variables), and the scope better delimited the variables didn't leave the scope.

but that has since changed, and it definitely makes sense to drop the scope here.